src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,68 @@
+package duke.task;
+
+import duke.task.ITask;
+
+public class Task implements ITask {
```

Comment:
Since only one task implements ITask, and Task holds some common implementations to all Tasks (markDone, getName, ...) you could consider making Task an abstract class instead.

Reply:
Thank you for the detailed and insightful comment, definitely learnt a lot :)! Greatly appreciate the time you took to look through my code.